### PR TITLE
Add `DataWrapperInteface`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  |

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: src/Rule/Count.php [hunk 1]
```diff
@@ -158,11 +156,4 @@ public function getHandler(): string
     {
         return CountHandler::class;
     }
-
-    public function afterInitAttribute(object $object, int $target): void
```
Comment (samdark) [inline src/Rule/Count.php:162]: Is it alright?

**Comment by samdark** [comment]
Diff should be re-checked.

File: src/Rule/Count.php (hunk 1 above)
Comment (vjik) [reply src/Rule/Count.php:162]: Yes. Now object itself passed to validation when rule applied to whole object.

**Comment by vjik** [comment]
> Diff should be re-checked.

Yes. Merge with master is done wrong.

**Comment by samdark** [comment]
Resolved conflicts, added docs.

**Pull request merged**